### PR TITLE
feat(piece): freshdesk 3 actions trying pull again

### PR DESCRIPTION
**Pull request by buttonsbond**
## What does this PR do?

Fixes # (issue)

### CODE REVIEW

File: packages/backend/.env
Comment (AbdulTheActivePiecer) [inline packages/backend/.env:1]: Please revert this file.

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            url,
+            headers,
+        };
+        const response = await httpClient.sendRequest(httprequestdata);
+
+        if (response.status == 200) {
+            const status=response.body.status;
+            const responderid=response.body.responder_id;
+            let ass='';
+            let tstatus='';
```
Comment (AbdulTheActivePiecer) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts]: Can you please rename these with a clearer name, I don't really get it

File: packages/pieces/freshdesk/src/lib/actions/get-tickets.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            'Authorization': FDapiToken,
+            'Content-Type' : 'application/json',
+        };
+
+       // const queryParams = new URLSearchParams();
+        //queryParams.append('number', context.propsValue.number || '');
+        //queryParams.append('client_id', context.propsValue.client_id || '');
+       // queryParams.append('project_id', context.propsValue.project_id || '');
+       // queryParams.append('description', context.propsValue.description || '');
+       // queryParams.append('rate', context.propsValue.rate?.toString() || '');
```
Comment (AbdulTheActivePiecer) [inline packages/pieces/freshdesk/src/lib/actions/get-tickets.ts]: Please remove these.

File: packages/ui/common/src/lib/environments/environment.ts
Comment (AbdulTheActivePiecer) [inline packages/ui/common/src/lib/environments/environment.ts:1]: Please revert this file.

**Review by AbdulTheActivePiecer — CHANGES_REQUESTED** [review]

**Comment by buttonsbond** [comment]
How do I revert the 2 files you specified please. I am using codespaces. 
I am pretty sure that the environment file was changed so that the 
frontend would work for me [i had the assistance of Mo on that]



------ Original Message ------
From "Abdul" ***@***.***>
To "activepieces/activepieces" ***@***.***>
Cc "Mark van Bellen" ***@***.***>; "Author" 
***@***.***>
Date 13/07/2023, 10:47:12
Subject Re: [activepieces/activepieces] feat(piece): freshdesk 3 actions 
trying pull again (PR #1642)

**Comment by AbdulTheActivePiecer** [comment]
@buttonsbond , oh he meant that locally only not to the main branch, because these environment changes are only related to the local development environment any contributor working on codespaces needs. basically this means, you can keep these changes locally but don't commit them.

**Comment by AbdulTheActivePiecer** [comment]
@abuaboud @khaledmashaly  I think we need to figure out a way to stop people from committing these files.

**Comment by AbdulTheActivePiecer** [comment]
@buttonsbond  to revert just paste the original files from here and commit:
https://github.com/activepieces/activepieces/blob/main/packages/backend/.env
https://github.com/activepieces/activepieces/blob/main/packages/ui/common/src/lib/environments/environment.ts

**Comment by buttonsbond** [comment]
I think that’s done. Sorry but I’m new to using GitHub in this way.



------ Original Message ------
From "Abdul" ***@***.***>
To "activepieces/activepieces" ***@***.***>
Cc "Mark van Bellen" ***@***.***>; "Mention" 
***@***.***>
Date 13/07/2023, 11:41:53
Subject Re: [activepieces/activepieces] feat(piece): freshdesk 3 actions 
trying pull again (PR #1642)

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+        const headers = {
+            'Authorization': FDapiToken,
+            'Content-Type' : 'application/json',
+        };
+
+        // Remove trailing slash from base_url
+        const baseUrl = context.auth.base_url.replace(/\/$/, "");
+        const url = `${baseUrl}/api/v2/tickets/${FDticketID}`;
+        const httprequestdata = {
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:32]:
```suggestion
        const httpRequestData = {
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    props: {
+        ticketid: Property.ShortText({
+            displayName: 'Ticket ID',
+            description: 'The Ticket ID to return status',
+            required: true,
+        }),
+    },
+
+    async run(context) {
+        const FDapiToken = context.auth.access_token;
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:21]:
```suggestion
        const fdApiToken = context.auth.access_token;
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        const FDticketID = context.propsValue.ticketid;
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:22]:
```suggestion
        const fdTicketID = context.propsValue.ticketid;
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+        const headers = {
+            'Authorization': FDapiToken,
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:25]:
```suggestion
            'Authorization': fdApiToken,
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            'Content-Type' : 'application/json',
+        };
+
+        // Remove trailing slash from base_url
+        const baseUrl = context.auth.base_url.replace(/\/$/, "");
+        const url = `${baseUrl}/api/v2/tickets/${FDticketID}`;
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:31]:
```suggestion
        const url = `${baseUrl}/api/v2/tickets/${fdTicketID}`;
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        const httprequestdata = {
+            method: HttpMethod.GET,
+            url,
+            headers,
+        };
+        const response = await httpClient.sendRequest(httprequestdata);
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:37]:
```suggestion
        const response = await httpClient.sendRequest(httpRequestData);
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+        if (response.status == 200) {
+            const status=response.body.status;
+            const responderid=response.body.responder_id;
+            let AssignedStatusFriendlyValue='';
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:42]:
```suggestion
            let assignedStatusFriendlyValue='';
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+            let TicketStatusFriendlyValue='';
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:43]:
```suggestion
            let ticketStatusFriendlyValue='';
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+            if (isNull(responderid)) {
+                AssignedStatusFriendlyValue="NOTASSIGNED";
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:46]:
```suggestion
                assignedStatusFriendlyValue="NOTASSIGNED";
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+            } else {
+                 AssignedStatusFriendlyValue="ASSIGNED";
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:48]:
```suggestion
                 assignedStatusFriendlyValue="ASSIGNED";
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+                case 3: { TicketStatusFriendlyValue=('PENDING'); break; }
+                case 4: { TicketStatusFriendlyValue=('RESOLVED'); break; }
+                case 5: { TicketStatusFriendlyValue=('CLOSED'); break; }
+                default: { 
+                    // if anything other than 2,3,4 or 5 just assign the value - it shouldn't happen!
+                    TicketStatusFriendlyValue=status;
+                    break; }
+            }
+            const json=[{
+                ticket_status: TicketStatusFriendlyValue,
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:62]:
```suggestion
                ticket_status: ticketStatusFriendlyValue,
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+                assigned_status: AssignedStatusFriendlyValue,
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:63]:
```suggestion
                assigned_status: assignedStatusFriendlyValue,
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+                assigned_id: responderid
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:64]:
```suggestion
                assigned_id: responderId
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+        const httprequestdata = {
+            method: HttpMethod.GET,
+            url,
+            headers,
+        };
+        const response = await httpClient.sendRequest(httprequestdata);
+
+        if (response.status == 200) {
+            const status=response.body.status;
+            const responderid=response.body.responder_id;
```
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:41]:
```suggestion
            const responderId = response.body.responder_id;
```

File: packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts (hunk 9 above)
Comment (JanHolger) [inline packages/pieces/freshdesk/src/lib/actions/get-ticket-status.ts:42]:
```suggestion
            let assignedStatusFriendlyValue = '';
```

**Pull request merged**